packages/shopping/src/spec-enhancers/info-spec.extension.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: loopback4-example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {bind, Context, inject} from '@loopback/context';
+import {
+  asSpecEnhancer,
+  mergeOpenAPISpec,
+  OASEnhancer,
+  OpenApiSpec,
+} from '@loopback/openapi-v3';
+import {PackageInfo, PackageKey} from '../application';
+
+@bind(asSpecEnhancer)
+export class InfoSpecEnhancer implements OASEnhancer {
+  name = 'info';
+  pkg: PackageInfo;
+
+  constructor(@inject.context() private ctx: Context) {
+    this.pkg = ctx.getSync<PackageInfo>(PackageKey);
```

Comment:
Why not `@inject(PakageKey) private pkg: PackageIno`?

Reply:
It complained at me when I tried that so I took path of last resistance and just rewrote. I'll redo and see if it was typescript engine being temperamental.

Reply:
@raymondfeng The error as follows:

```node
UnhandledPromiseRejectionWarning: Error: Cannot resolve injected arguments for InfoSpecEnhancer.[0]: The arguments[0] is not decorated for dependency injection, but a value is not supplied
[0]     at /home/dev/workspace/loopback4-example-shopping/packages/shopping/node_modules/@loopback/context/dist/resolver.js:167:23
```

Reply:
It's weird. I'll troubleshoot.

Reply:
See https://github.com/strongloop/loopback4-example-shopping/pull/611/files#r396672461